### ISSUE TITLE
7 - Weird refresh problem 

### ISSUE THREAD

**Issue by arthur-debert** [arthur-debert/BulkLoader#7]
<pre>
<b>What steps will reproduce the problem?</b>

Situation:
I have one 200k swf which I am preloading with BulkLoader. This swf contains 4 movieclips 
which were done in Illustrator (have some complex lines but it shouldn't matter). I am loading 
this swf from a main movie with the method described in the help file. I am using Flash IDE with 
AS3.

The problem:
When I export the swfs and test it in the browser tracing the PROGRESS event produces what is 
expected:
loaded x of y bytes. where x starts from 0 and reaches the total number of bytes gradually in a 
number of steps.
Hovewer when it finished loading when I add the swf to the display list I can only see one mc 
inside the loaded swf, sometimes none (intermittent).

The weird thing is that, if now I refresh my browser and do the whole procedure again I get the 
following tracing the progress (seems it's loading from cache):
loaded 0 of 201835
loaded 201835 of 201835
loaded 201835 of 201835

Then everything displays correctly. 

It is always the first load when I have correct trace but incorrect display, then incorrect trace and 
correct display.

I tried my code using the Loader class and everything works perfectly so that's why I suspect 
that there must be something in Bulk Loader which produces this. 

<b>What version of the product are you using? On what operating system?</b>
0.9.2.1 rev73 on OSX 10.5 Safari, but tested on Firefox as well.

Great class nevertheless, just this strange issue.

Oszkar Nagy

</pre></br>Original link: http://code.google.com/p/bulk-loader/issues/detail?id=7

**Comment by arthur-debert** [comment]
debert - 2007-12-11 00:00:00 

Hi Oskzar.

This is pretty strange, I load swfs with it a lot, and no issues.
This looks like your swf has some sort of code on it that depends on loading to show
all mcs. The first time it varies (because loading is unpredictable), but when in
cache it won't fail.

The progress trace patterns are correct. When in caches bytesLoaded is 0 until the
player is told the file is in cache, and then since it's reading from disk, it reads
200k instantly, and you get 100% the next step.

The best way to test this is to produce a new swf to be loaded, no code, just on
large image on stage for example, use your loading code to load that.

If you can't figure this out, can you make a zip file with the minimum reproducible
setup, so I can debug this?

Thanks
Arthur

**Comment by arthur-debert** [comment]
oszkar.nagy - 2007-12-11 00:00:00 

Hi Arthur!

Thanks for reply. I did some more experiments, and you are probably right as I couldn't reproduce the 
problem in a clean test situation. I isolated the loader code and I loaded an swf containing the same graphics 
and it loaded up perfectly, with the progress traces being what I expected and all the graphics showed up.

I will check my code further as I suspect there must be a bug somewhere which is causing this (it's a much 
larger app so a lot of things can go wrong). If I find anything which cleanly traces back to the loader I'll let you 
know and/or if I find anything in connection with other stuff and the loader, maybe useful for others. 

Thanks for your help, and all the best.
Oszkar
